common/djangoapps/student/tasks.py

```diff
@@ -0,0 +1,61 @@
+"""
+This file contains celery tasks for sending email
+"""
+import logging
+from django.conf import settings
+from django.core import mail
+
+from celery.task import task  # pylint: disable=no-name-in-module, import-error
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+from edxmako.shortcuts import render_to_string
+
+log = logging.getLogger("edx.student")
+
+
+@task()
+def send_activation_email(user, activation_key):
+
+    """
+        Sending an activation email to the users.
+    """
+    context = {
+        'name': user.profile.name,
+        'key': activation_key,
+    }
+    # composes activation email
+    subject = render_to_string('emails/activation_email_subject.txt', context)
+    # Email subject *must not* contain newlines
+    subject = ''.join(subject.splitlines())
+    message = render_to_string('emails/activation_email.txt', context)
+
+    from_address = configuration_helpers.get_value(
+        'email_from_address',
+        settings.DEFAULT_FROM_EMAIL
+    )
+    retries = settings.RETRY_ACTIVATION_EMAIL_MAX_ATTEMPTS
+    dest_addr = user.email
+    while retries:
```

Comment:
I know, you're right. I'm just always scared my while loops will result in an infinite loop 😨